packages/vite/src/node/config.ts

```diff
@@ -354,6 +356,13 @@ export async function resolveConfig(
     }
   }
 
+  let publicDir = config.publicDir
+  if (typeof publicDir === 'boolean') {
+    publicDir = publicDir ? path.resolve(resolvedRoot, 'public') : ''
+  } else {
+    publicDir = path.resolve(resolvedRoot, publicDir || 'public')
+  }
```

Comment:
This looks good, and it is according to what we discussed. I am thinking though if we shouldn't directly break already the undocumented behaviour of `publicDir ==== ''`. As it is quite confusing that:
a. if publicDir is '' in UserConfig, it means public
b. if publicDir is '' in ResolvedConfig, it means disabled

@antfu @Shinigami92 what do you think?

Example proposed change if we decide to already change it so '' always means disabled.
```suggestion
  const { publicDir } = config
  let resolvedPublicDir = publicDir === false || publicDir === '' ? '' : 
    path.resolve(resolvedRoot, typeof publicDir === 'string ? publicDir : 'public' )
```
  
  }
```
  
  path.resolve(resolvedRoot, 
      publicDir === true ? 'public'  publicDir === '' ? 'public' : publicDir
  )
  }
```

Reply:
IMO the best case would be to show the dev an error when they insert an _invalid_ dir like `''`  
That could prevent further issues and doesn't lead to confusions like:

> "I set it to '', why does it take 'public'? :thinking:"

Reply:
Probably an error is the best way to go here, you are right. Ok, let's give Evan these options 👍🏼